### PR TITLE
[BPK-1353] Add description prop to native text input

### PR DESCRIPTION
**Pull request by shaundon**
## To do
* [x] Implementation
* [x] Storybook
* [x] Tests
* [x] Readme
* [x] Changelog
* [x] Screenshots

### With no value
screen shot 2018-02-22 at 15 09 57

### With a value
screen shot 2018-02-22 at 15 10 13

### Focused
screen shot 2018-02-22 at 15 10 05

### In context with other fields around it
screen shot 2018-02-22 at 15 10 20

### CODE REVIEW

File: native/packages/react-native-bpk-component-text-input/stories.js [hunk 1]
```diff
@@ -71,13 +71,14 @@ storiesOf('BpkTextInput', module).add('docs:text-inputs', () => (
...
     <StatefulBpkTextInput
-      label="Input with value"
+      label="Input with value and description"
       initialValue="Edinburgh"
+      description="What's the capital of Scotland?"
       style={styles.input}
     />
     <StatefulBpkTextInput
       label="Input with multiline value"
-      initialValue="Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus." // eslint-disable-line max-len
```
Comment (shaundon) [inline native/packages/react-native-bpk-component-text-input/stories.js]: Removed some of this to save space in the screenshots.

File: native/packages/react-native-bpk-component-text-input/readme.md [hunk 2]
```diff
@@ -41,6 +41,7 @@ export default class App extends Component {
           label="City"
           style={styles.input}
           value="Edinburgh"
+          description="What's the capital of Scotland?"
```
Comment (georgegillams) [inline native/packages/react-native-bpk-component-text-input/readme.md]: Is this really how we want the `description` prop to be used?

File: native/packages/react-native-bpk-component-text-input/src/BpkTextInput.js [hunk 3]
```diff
@@ -162,6 +166,24 @@ class BpkTextInput extends Component<Props, State> {
       valid,
     );
 
+    const renderExtraInfo = () => {
```
Comment (georgegillams) [inline native/packages/react-native-bpk-component-text-input/src/BpkTextInput.js]: I'm not sure that there is any benefit to this being a separate function. Maybe some readability gain from not having it all in-line, but it's not a massive component anyway

File: native/packages/react-native-bpk-component-text-input/readme.md (hunk 2 above)
Comment (shaundon) [reply native/packages/react-native-bpk-component-text-input/readme.md]: No, I was being whimsical but I guess it's better to lead by example and show a useful message! I'll change it.

File: native/packages/react-native-bpk-component-text-input/src/BpkTextInput.js (hunk 3 above)
Comment (shaundon) [reply native/packages/react-native-bpk-component-text-input/src/BpkTextInput.js]: Hmm yeah, I see your point, but to me the inline version is less readable:

```
        {valid === false &&
          validationMessage && (
            <BpkText textStyle="xs" style={styles.validationMessage}>
              {validationMessage}
            </BpkText>
          )}
        {valid !== false &&
          description && (
            <BpkText textStyle="xs" style={styles.description}>
              {description}
            </BpkText>
          )}
```

What do you think? cc @matteo-hertel @matthewdavidson @k0nserv

Comment (matthewdavidson) [reply native/packages/react-native-bpk-component-text-input/src/BpkTextInput.js]: I think it's fine as is 👍

Comment (matteo-hertel) [reply native/packages/react-native-bpk-component-text-input/src/BpkTextInput.js]: It makes the render method more readable, so gest a 👍 from me!

**Pull request merged**